odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1064 @@
+                'author': self.user.id,
+                'custom_properties': {
+                    'name': 'Test',  # do not exists on discussion 2, should be removed
+                },
+            }])
+
+        self.assertEqual(len(messages), 2)
+
+        properties_values_1 = messages[0].custom_properties
+        properties_values_2 = messages[1].custom_properties
+
+        self.assertEqual(len(properties_values_1), 2, msg='Discussion 1 has 2 properties')
+        self.assertEqual(len(properties_values_2), 1, msg='Discussion 2 has 1 properties')
+
+        self.assertEqual(properties_values_1[0]['value'], 'Test')
+        self.assertEqual(properties_values_1[1]['value'], self.partner.id)
+        self.assertEqual(properties_values_2[0]['value'], 'draft', msg='Should have take the default value')
+
+    def test_properties_field_default(self):
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'discussion': self.discussion_2.id,
+            'author': self.user.id,
+        })
+        self.assertEqual(
+            message.custom_properties[0]['value'],
+            'draft',
+            msg='Should have take the default value')
+
+        message.custom_properties = {'state': None}
+        self.assertFalse(
+            message.custom_properties[0]['value'],
+            msg='Writing None should not reset the default value')
+
+        # test the case where the parent come from a default as well
+        self.env['test_new_api.message']._fields['discussion'].default = lambda __: self.discussion_2.id
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'author': self.user.id,
+        })
+        self.assertEqual(message.discussion, self.discussion_2)
+        self.assertEqual(
+            message.custom_properties[0]['value'],
+            'draft',
+            msg='Should have take the default value')
+
+        # the parent come from a default value
+        self.discussion_2.message_properties = [{'id': 'test', 'type': 'char', 'default': 'default char'}]
+        default_values = self.env['test_new_api.message'].default_get(
+            ['discussion', 'custom_properties'])
+        self.assertEqual(default_values.get('discussion'), self.discussion_2.id)
+        self.assertEqual(
+            default_values.get('custom_properties', {}).get('test'),
+            'default char')
+
+        # test a default many2one
+        self.discussion_1.message_properties = [
+            {
+                'id': 'my_many2one',
+                'string': 'Partner',
+                'model': 'res.partner',
+                'type': 'many2one',
+                # send the value like the web client does
+                'default': [self.partner.id, 'Bob'],
+            },
+        ]
+        sql_definition = self._get_sql_definition(self.discussion_1)
+        self.assertEqual(sql_definition[0]['default'], self.partner.id)
```

Comment:
Looks like we don't store it after all? Then not sure why it's specified here above.

Don't forget that the web client is the client of the server, and not the opposite, not everything should be tailored to fit what the web client sends/receives.